x-pack/plugins/ml/server/models/results_service/results_service.ts

```diff
@@ -605,6 +612,104 @@ export function resultsServiceProvider(mlClient: MlClient) {
     return finalResults;
   }
 
+  async function getDatafeedResultsChartData({
+    jobId,
+    start,
+    end,
+  }: DatafeedResultsChartDataParams): Promise<GetDatafeedResultsChartDataResult> {
+    const finalResults: GetDatafeedResultsChartDataResult = {
+      bucketResults: [],
+      datafeedResults: [],
+    };
+
+    const { getDatafeedByJobId } = datafeedsProvider(client!, mlClient);
+    const datafeedConfig = await getDatafeedByJobId(jobId);
+
+    const {
+      body: { jobs },
+    } = await mlClient.getJobs({ job_id: jobId });
+    const jobConfig = jobs[0];
```

Comment:
`jobs[0]` could be `undefined` here if this endpoint is called with a wildcard in the `jobId`
Testing this throws a kibana internal server error.

Reply:
Good catch - updated to handle the job not being found in 1df1594527f9d947920775b41019c80641ca2777